ext/js/dictionary/dictionary-importer.js

```diff
@@ -691,18 +691,19 @@ export class DictionaryImporter {
     _getArchiveFiles(fileMap, queryDetails) {
         /** @type {import('dictionary-importer').QueryResult} */
         const results = new Map();
-        for (const [name, value] of fileMap.entries()) {
-            for (const [fileType, fileNameFormat] of queryDetails.entries()) {
-                let entries = results.get(fileType);
-                if (typeof entries === 'undefined') {
-                    entries = [];
-                    results.set(fileType, entries);
-                }
 
-                if (fileNameFormat.test(name)) {
-                    entries.push(value);
-                    break;
-                }
+        for (const [fileType] of queryDetails) {
```

Comment:
`for (const fileType of queryDetails.keys())`

Reply:
I changed `queryDetails` from Map type to 2D array type, so `queryDetails.keys()` would return the indexes 0, 1, 2, 3, 4. Or are you thinking of something else?

Reply:
@toasted-nutbread

Reply:
`for (const [k, v] of map) { ... }` is what you are implicitly doing, just omitting the `v`. If you want to iterate over only the keys, just iterate over the `.keys()` iterable using `for (const k of map.keys())`

Reply:
Wait no I'm dumb again because I wasn't reading the diff in context. ignore me.